### ISSUE TITLE
Invalid credentials warning in Hazelcast 3.3.4

### ISSUE THREAD

**Issue by dsukhoroslov** [hazelcast/hazelcast#4449]
Hazelcast 3.3.4. Client connects to the server:
```
ClientConfig config = new XmlClientConfigBuilder(in).build();
config.getGroupConfig().setName(schema);
config.getGroupConfig().setPassword(password);
config.getNetworkConfig().addAddress(address);
hzClient = HazelcastClient.newHazelcastClient(config);
```

at this point the following warning is logged:
```
янв 20, 2015 11:09:51 AM com.hazelcast.core.LifecycleService
INFO: HazelcastClient[hz.client_0_TPoX2][3.3.4] is STARTING
янв 20, 2015 11:09:53 AM com.hazelcast.core.LifecycleService
INFO: HazelcastClient[hz.client_0_TPoX2][3.3.4] is STARTED
янв 20, 2015 11:09:53 AM com.hazelcast.client.spi.impl.ClusterListenerThread
WARNING: Exception during initial connection to Address[127.0.0.1]:5701
com.hazelcast.client.AuthenticationException: Invalid credentials!
	at com.hazelcast.client.impl.client.AuthenticationRequest.handleUnauthenticated(AuthenticationRequest.java:128)
	at com.hazelcast.client.impl.client.AuthenticationRequest.call(AuthenticationRequest.java:72)
	at com.hazelcast.client.impl.client.CallableClientRequest.process(CallableClientRequest.java:29)
	at com.hazelcast.client.impl.ClientEngineImpl$ClientPacketProcessor.processRequest(ClientEngineImpl.java:439)
	at com.hazelcast.client.impl.ClientEngineImpl$ClientPacketProcessor.run(ClientEngineImpl.java:351)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:92)
	at ------ End remote and begin local stack-trace ------.(Unknown Source)
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl.sendAndReceive(ClientConnectionManagerImpl.java:505)
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl.authenticate(ClientConnectionManagerImpl.java:479)
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl.access$2100(ClientConnectionManagerImpl.java:87)
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl$ManagerAuthenticator.auth(ClientConnectionManagerImpl.java:458)
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl$ConnectionProcessor.call(ClientConnectionManagerImpl.java:400)
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl$ConnectionProcessor.call(ClientConnectionManagerImpl.java:358)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at com.hazelcast.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:57)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:92)
Caused by: com.hazelcast.client.AuthenticationException: Invalid credentials!
	at com.hazelcast.client.impl.client.AuthenticationRequest.handleUnauthenticated(AuthenticationRequest.java:128)
	at com.hazelcast.client.impl.client.AuthenticationRequest.call(AuthenticationRequest.java:72)
	at com.hazelcast.client.impl.client.CallableClientRequest.process(CallableClientRequest.java:29)
	at com.hazelcast.client.impl.ClientEngineImpl$ClientPacketProcessor.processRequest(ClientEngineImpl.java:439)
	at com.hazelcast.client.impl.ClientEngineImpl$ClientPacketProcessor.run(ClientEngineImpl.java:351)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)
	... 2 more

янв 20, 2015 11:09:53 AM com.hazelcast.core.LifecycleService
INFO: HazelcastClient[hz.client_0_TPoX2][3.3.4] is CLIENT_CONNECTED
янв 20, 2015 11:09:53 AM com.hazelcast.client.spi.impl.ClusterListenerThread
INFO: 

Members [1] {
	Member [localhost]:10500
}
```

but client is connected and works properly. On the server side the log is:
```
[java] 2015-01-20 11:09:53.638 [hz.hzInstance.IO.thread-Acceptor] INFO  com.hazelcast.nio.tcp.SocketAcceptor - [localhost]:5701 [system] [3.3.4] Accepting socket connection from /127.0.0.1:59243
[java] 2015-01-20 11:09:53.638 [hz.hzInstance.cached.thread-2] INFO  com.hazelcast.nio.tcp.TcpIpConnectionManager - [localhost]:5701 [system] [3.3.4] Established socket connection between /127.0.0.1:5701 and /127.0.0.1:59243
[java] 2015-01-20 11:09:53.653 [hz.hzInstance.client.thread-7] WARN  com.hazelcast.client.impl.client.AuthenticationRequest - [localhost]:5701 [system] [3.3.4] Received auth from Connection [/127.0.0.1:5701 -> /127.0.0.1:59243], endpoint=null, live=true, type=JAVA_CLIENT, authentication failed
[java] 2015-01-20 11:09:53.669 [hz.hzInstance.IO.thread-in-2] INFO  com.hazelcast.nio.tcp.TcpIpConnection - [localhost]:5701 [system] [3.3.4] Connection [/127.0.0.1:59243] lost. Reason: java.io.EOFException[Remote socket closed!]
[java] 2015-01-20 11:09:53.669 [hz.hzInstance.IO.thread-in-2] WARN  com.hazelcast.nio.tcp.ReadHandler - [localhost]:5701 [system] [3.3.4] hz.hzInstance.IO.thread-in-2 Closing socket to endpoint null, Cause:java.io.EOFException: Remote socket closed!
[java] 2015-01-20 11:09:53.685 [hz.TPoX2.IO.thread-Acceptor] INFO  com.hazelcast.nio.tcp.SocketAcceptor - [localhost]:10500 [TPoX2] [3.3.4] Accepting socket connection from /127.0.0.1:59246
[java] 2015-01-20 11:09:53.685 [hz.TPoX2.cached.thread-4] INFO  com.hazelcast.nio.tcp.TcpIpConnectionManager - [localhost]:10500 [TPoX2] [3.3.4] Established socket connection between /127.0.0.1:10500 and /127.0.0.1:59246
[java] 2015-01-20 11:09:53.685 [hz.TPoX2.client.thread-121] INFO  com.hazelcast.client.impl.client.AuthenticationRequest - [localhost]:10500 [TPoX2] [3.3.4] Received auth from Connection [/127.0.0.1:10500 -> /127.0.0.1:59246], endpoint=null, live=true, type=JAVA_CLIENT, successfully authenticated
[java] 2015-01-20 11:09:53.700 [hz.TPoX2.IO.thread-Acceptor] INFO  com.hazelcast.nio.tcp.SocketAcceptor - [localhost]:10500 [TPoX2] [3.3.4] Accepting socket connection from /127.0.0.1:59249
[java] 2015-01-20 11:09:53.700 [hz.TPoX2.cached.thread-4] INFO  com.hazelcast.nio.tcp.TcpIpConnectionManager - [localhost]:10500 [TPoX2] [3.3.4] Established socket connection between /127.0.0.1:10500 and /127.0.0.1:59249
[java] 2015-01-20 11:09:53.700 [hz.TPoX2.client.thread-111] INFO  com.hazelcast.client.impl.client.AuthenticationRequest - [localhost]:10500 [TPoX2] [3.3.4] Received auth from Connection [/127.0.0.1:10500 -> /127.0.0.1:59249], endpoint=null, live=true, type=JAVA_CLIENT, successfully authenticated
```

There are two Hazelcast instances started on the server side, one on the default port 5701 and the second at port 10500. I do connect to the second instance only. But it looks like the client tries to connect at default port anyway. There is no such issue in Hazelcast 3.3.3

**Comment by bwzhang2011** [comment]
@gurbuzali, is such issue confirm. hope such should be fixed with some patch for 3.3.4 for our project has upgraded to such version.

**Comment by dsukhoroslov** [comment]
The workaround is simple: just do not start Hazelcast server on the default port. But this slows down client connection significantly: initially it tries to connect on default port and only after that it uses the address provided in config.

**Comment by bwzhang2011** [comment]
@dsukhoroslov, thanks a lot for feed back but I could not quite understand what you pointed out. as we know, the default port is just for test and development, we will change that to proper one. and what's more, how or why would it slow down the connection according to the configuration of hazelcast client and in our project we rely to hazelcast spring configuration. so @gurbuzali , once it was confirmed hope the patch would be provided not long in the future or released in 3.3.5

**Comment by dsukhoroslov** [comment]
Sorry, that was my bad. The default connection settings were specified in my default config file and then the correct connection address was added dynamically. Because connections shuffle before the first use and some bad luck I never seen this warning at Hazelacast 3.3.3, but found it at Hazelcast 3.3.4. So, the issue is closed.
